src/rednoise.c

```diff
@@ -3,12 +3,20 @@
 #include "rednoise_cmd.h"
 #ifdef USEDMALLOC
 #include "dmalloc.h"
+#include "misc_utils.h"
 #endif
 
+#include <sys/types.h>
+#include <sys/stat.h>
+struct stat info;
+
 int main(int argc, char *argv[])
 {
     FILE *infile, *outfile;
     char *rootfilenm, *outname;
+    char *fftfullpath;
```

Comment:
Note that you never free these.  Not a critical problem since they will get freed when the program exits, but it is good form to free them when you are finished with them.

Reply:
Right, good point! I can see that I am not used to program in C/C++ anymore... I will add the free() call for the sake of correctness.